subsys/mgmt/mcumgr/smp_bt.c

```diff
@@ -89,62 +103,48 @@ static struct conn_param_data *get_conn_param_data(const struct bt_conn *conn)
 }
 
 /* Sets connection parameters for a given conn. */
-static void set_conn_latency(struct bt_conn *conn, bool low_latency)
+static void conn_params_set(struct bt_conn *conn, bool smp_params)
 {
-	struct bt_le_conn_param params;
-	struct conn_param_data *cpd;
-	struct bt_conn_info info;
 	int ret = 0;
+	const struct bt_le_conn_param *params;
+	struct conn_param_data *cpd;
 
 	cpd = get_conn_param_data(conn);
+	params = smp_params ? SMP_BT_CONN_PARAMS : DEFAULT_CONN_PARAMS;
 
-	ret = bt_conn_get_info(conn, &info);
-	__ASSERT_NO_MSG(!ret);
-
-	if ((low_latency && (info.le.latency == 0)) ||
-	    ((!low_latency) && (info.le.latency == DEFAULT_LATENCY))) {
-		/* Already updated. */
-		return;
-	}
-
-	params.interval_min = info.le.interval;
-	params.interval_max = info.le.interval;
-	params.latency = (low_latency) ? (0) : (DEFAULT_LATENCY);
-	params.timeout = info.le.timeout;
-
-	ret = bt_conn_le_param_update(cpd->conn, &params);
+	ret = bt_conn_le_param_update(conn, params);
 	if (ret && (ret != -EALREADY)) {
-		if (!low_latency) {
-			/* Try again to avoid stucking in low latency. */
+		if (!smp_params) {
+			/* Try again to avoid stucking in SMP connection parameters. */
 			(void)k_work_reschedule(&cpd->dwork, K_MSEC(RESTORE_RETRY_TIME));
 		}
 	}
 }
 
 
-/* Work handler function for restoring the default latency for the connection. */
-static void restore_default_latency(struct k_work *work)
+/* Work handler function for restoring the default connection parameters for the connection. */
+static void pref_conn_params_restore(struct k_work *work)
 {
 	struct conn_param_data *cpd;
 
 	cpd = CONTAINER_OF(work, struct conn_param_data, dwork);
 
-	if (cpd->latency_state & CONN_LOW_LATENCY_REQUIRED) {
-		cpd->latency_state &= ~CONN_LOW_LATENCY_REQUIRED;
+	if (cpd->state & SMP_CONN_PARAMS_REQUIRED) {
+		cpd->state &= ~SMP_CONN_PARAMS_REQUIRED;
 		(void)k_work_reschedule(&cpd->dwork, K_MSEC(RESTORE_TIME));
 	} else {
-		set_conn_latency(cpd->conn, false);
+		conn_params_set(cpd->conn, false);
 	}
 }
 
-static void enable_low_latency(struct bt_conn *conn)
+static void smp_conn_params_enable(struct bt_conn *conn)
 {
 	struct conn_param_data *cpd = get_conn_param_data(conn);
 
-	if (cpd->latency_state & CONN_LOW_LATENCY_ENABLED) {
-		cpd->latency_state |= CONN_LOW_LATENCY_REQUIRED;
+	if (cpd->state & SMP_CONN_PARAMS_ENABLED) {
+		cpd->state |= SMP_CONN_PARAMS_REQUIRED;
 	} else {
-		set_conn_latency(conn, true);
+		conn_params_set(conn, true);
```

Comment:
If connected peer is Central and it would not provide us all of the requested parameters, we may flood the Central with reqests. E.g. we may require interval of 7.5 which is never provided by an Android device.